src/api/spec/db/data/remove_duplicate_repositories_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+require Rails.root.join('db/data/20170306084550_remove_duplicate_repositories.rb')
+require Rails.root.join('db/migrate/20170306084558_change_repositories_remote_project_name_to_not_null.rb')
+
+RSpec.describe RemoveDuplicateRepositories, type: :migration do
+  # This migration does not allow NULL values for remote_project_name column
+  # We need this migration to not be run to create test data
+  # 20170306084558_change_repositories_remote_project_name_to_not_null.rb
+  let(:schema_migration_version) { 20_170_306_084_558 }
+  let(:schema_migration) { ChangeRepositoriesRemoteProjectNameToNotNull.new }
+  let(:data_migration_version) { 20_170_306_084_550 }
+  let(:data_migration) { RemoveDuplicateRepositories.new }
+
+  describe '.up' do
+    before do
+      schema_migration.down
+    end
+
+    after do
+      DataMigrate::DataSchemaMigration.find_or_create_by(version: data_migration_version.to_s)
+      schema_migration.up
+    end
+
+    let(:remote_project) { create(:remote_project) }
+    let!(:remote_repository) { create(:repository, project: remote_project, name: 'standard', remote_project_name: 'openSUSE.org:Foo') }
+    let!(:remote_repository_with_same_name) do
+      create(:repository, project: remote_project, name: 'standard', remote_project_name: 'openSUSE.org:Bar')
+    end
+
+    let(:project) { create(:project) }
+    let!(:repository) do
+      # we need to bypass the validation
+      r = build(:repository, project: project, name: 'standard', remote_project_name: nil, architectures: ['x86_64', 'i586'])
+      r.save(validate: false)
+      r
+    end
+    let!(:duplicate_repository) do
+      # we need to bypass the validation
+      r = build(:repository, project: project, name: 'standard', remote_project_name: nil, architectures: ['x86_64', 'i586'])
+      r.save(validate: false)
+      r
+    end
+    # We need to set remote_project_name = '', otherwise Repository.deleted_instance will fail because of not nil validation
+    let!(:deleted_repository) { create(:repository, name: 'deleted', remote_project_name: '', project: Project.deleted_instance) }
```

Comment:
Can you move this up a couple of lines? Before all the other data? I think the comment should also explain a bit more why this is needed. In 2 weeks you won't remember what `Repository.deleted_instance` does.